chemprop/uncertainty/uncertainty_evaluator.py

```diff
@@ -0,0 +1,505 @@
+from typing import List
+
+import numpy as np
+from scipy.stats import t, spearmanr
+from scipy.special import erfinv
+
+from chemprop.data import MoleculeDataset
+from .uncertainty_calibrator import UncertaintyCalibrator
+from chemprop.train import evaluate_predictions
+
+
+class UncertaintyEvaluator:
+    """
+    A class for evaluating the effectiveness of uncertainty estimates with metrics.
+    """
+
+    def __init__(
+        self,
+        evaluation_method: str,
+        calibration_method: str,
+        uncertainty_method: str,
+        dataset_type: str,
+        loss_function: str,
+        calibrator: UncertaintyCalibrator,
+    ):
+        self.evaluation_method = evaluation_method
+        self.calibration_method = calibration_method
+        self.uncertainty_method = uncertainty_method
+        self.dataset_type = dataset_type
+        self.loss_function = loss_function
+        self.calibrator = calibrator
+
+    def raise_argument_errors(self):
+        """
+        Raise errors for incompatibilities between dataset type and uncertainty method, or similar.
+        """
+        if self.dataset_type == "spectra":
+            raise NotImplementedError(
+                "No uncertainty evaluators implemented for spectra dataset type."
+            )
+
+    def evaluate(
+        self,
+        test_data: MoleculeDataset,
+        preds: List[List[float]],
+        uncertainties: List[List[float]],
+    ):
+        """
+        Run evaluation of predicted uncertainty values
+        """
+        pass
+
+
+class MetricEvaluator(UncertaintyEvaluator):
+    """
+    A class for evaluating confidence estimates of classification and multiclass datasets using builtin evaluation metrics.
+    """
+
+    def __init__(
+        self,
+        evaluation_method: str,
+        calibration_method: str,
+        uncertainty_method: str,
+        dataset_type: str,
+        loss_function: str,
+        calibrator: UncertaintyCalibrator,
+    ):
+        super().__init__(
+            evaluation_method=evaluation_method,
+            calibration_method=calibration_method,
+            uncertainty_method=uncertainty_method,
+            dataset_type=dataset_type,
+            loss_function=loss_function,
+            calibrator=calibrator,
+        )
```

Comment:
```suggestion
```
all this is doing is calling the superclass's `init()` function with the same argument. There's no need to define a subclass `init()` if that's all it does because that's the default behavior. This goes for every subclass defined below as well

Reply:
Yep, you're right. I added them to all of them assuming that there would be unique top line activities that I would have to do for all of them, but then that ended up not being the case. Will remove.